teuthology/task/ssh_keys.py

```diff
@@ -154,13 +154,13 @@ def push_keys_to_host(ctx, config, public_key, private_key):
             priv_key_data = '{priv_key}'.format(priv_key=private_key)
             misc.delete_file(remote, priv_key_file, force=True)
             # Hadoop requires that .ssh/id_rsa have permissions of '500'
-            misc.create_file(remote, priv_key_file, priv_key_data, str(500))
+            remote.write_file(priv_key_file, priv_key_data, mode='0500')
```

Comment:
could you explain the reason why we need this change in the commit message?

Reply:
I have updated the commit message.